packages/shared/src/components/modals/RanksModal/DevCardFooter.tsx

```diff
@@ -20,17 +31,29 @@ export default function DevCardFooter({
           isLocked={isLocked}
         />
         <div className="flex flex-col flex-1 items-start ml-6">
-          <ModalText>
-            DevCard is your developer ID. It showcases your achievements to the
-            world.
-          </ModalText>
-          <GoToDevCardButton
-            origin="ranks instructions"
-            className="mt-3"
-            isLocked={isLocked}
-          >
-            Generate
-          </GoToDevCardButton>
+          <ModalText>{devCardText({ user, isLocked })}</ModalText>
+          {user && isLocked ? (
+            <div className="relative mt-2 flex w-full items-center">
+              <strong className="typo-footnote">
+                {reads}/{devCardLimit}
+              </strong>
+              <div className="ml-2 flex flex-1 relative h-2 rounded-full bg-theme-active">
+                <div
+                  className="absolute bottom-0 left-0 h-full rounded-full bg-theme-label-primary"
+                  style={{ width: (100 * reads) / devCardLimit }}
+                  data-testId="tagProgress"
+                />
+              </div>
+            </div>
+          ) : (
+            <GoToDevCardButton
```

Comment:
@rebelchris In every scenario, in the design, the devcard button is always rendered. Why do we have to conditionally render it?

Reply:
See above comment:
https://www.sketch.com/s/2867ec63-0966-4647-b1d7-b6fe696ba251/a/jgA5lWz

Reply:
Nice. Then all the changes were valid. Thanks for the help!